cloud/vmware/vsphere_guest.py

```diff
@@ -858,6 +858,50 @@ def reconfigure_vm(vsphere_client, vm, module, esxi, resource_pool, cluster_name
 
             changes['cpu'] = vm_hardware['num_cpus']
 
+    # Resize hard drives
+    if vm_disk:
+        spec = spec_singleton(spec, request, vm)
+        disk_num = 0
+        dev_changes = []
+        disks_changed = {}
+        for disk in sorted(vm_disk.iterkeys()):
+            try:
+                disksize = int(vm_disk[disk]['size_gb'])
+                # Convert the disk size to kilobytes
+                disksize = disksize * 1024 * 1024
+            except (KeyError, ValueError):
+                vsphere_client.disconnect()
+                module.fail_json(msg="Error in '%s' definition. Size needs to be specified as an integer." % disk)
+
+            # Get a list of the hard drives 
+            dev_list = [d for d in vm.properties.config.hardware.device if d._type=='VirtualDisk']
```

Comment:
It may be more efficient to create `dev_list` prior to the `for disk in...` loop.